parsers/wiki_parser.py

```diff
@@ -0,0 +1,291 @@
+        processed_text = clean_response_text(original_text)
+        if processed_text != '':
+            try:
+                link = file_and_link_dict[file]
+                responses_list.append((original_text, processed_text, link))
+            except KeyError:
+                # Ignore files with no links to mp3 files. Happens to broken files and files undergoing migration.
+                pass
+
+    return responses_list
+
+
+def parse_response(text):
+    # Special cases
+    if '(broken file)' in text:
+        return ''
+    if 'versus (TI ' in text:
+        return ''
+    if 'Ceeeb' in text:
+        return ''
+
+    text = re.sub(r'…', '...', text)  # Replace ellipsis with three dots
+
+    regexps_empty_sub = [r'<!--.*?-->',  # Remove comments
+                         r'{{resp\|(r|u|\d+|d\|\d+)}}',  # Remove response rarity
+                         r'{{hero icon\|[a-z- \']+\|\d+px}}',  # Remove hero icon
+                         r'{{item( icon)?\|[a-z0-9() \']+\|\d+px}}',  # Remove item icon
+                         r'\[\[File:[a-z.,!\'() ]+\|\d+px\|link=[a-z,!\'() ]+]]',  # Remove Files
+                         r'<small>\[\[#[a-z_\-\' ]+\|\'\'followup\'\']]</small>',
+                         # Remove followup links in <small> tags
+                         r'<small>\'\'[a-z0-9 /]+\'\'</small>',  # Remove text in <small> tags
+                         r'<ref>.*?</ref>',  # Remove text in <ref> tags
+                         r'<nowiki>.*?</nowiki>',  # Remove text in <nowiki> tags
+                         ]
+    for regex in regexps_empty_sub:
+        text = re.sub(regex, '', text, flags=re.IGNORECASE)
+
+    regexps_sub_text = [r'\[\[([a-zé().:\',\- ]+)]]',  # Replace links such as [[Shitty Wizard]]
+                        r'\[\[[a-zé0-9().:\'/ ]+\|([a-zé().:\' ]+)]]',
+                        # Replace links such as  [[Ancient (Building)|Ancients]] and [[:File:Axe|Axe]]
+                        # r'{{h\|([a-zé().:\' ]+)}}',  # Replace hero names
+                        r'{{tooltip\|(.*?)\|.*?}}',  # Replace tooltips
+                        r'{{note\|([a-z.!\'\-?, ]+)\|[a-z.!\'\-?,()/ ]+}}',  # Replace notes
+                        ]
+    for regex in regexps_sub_text:
+        text = re.sub(regex, '\\1', text, flags=re.IGNORECASE)
+
+    if any(escape in text for escape in ['[[', ']]', '{{', '}}', '|', 'sm2']):
+        logger.warn('Response could not be processed : ' + text)
+
+    return text.strip()
+
+
+def clean_response_text(response_text):
+    """Method for preprocessing the given response text.
+    It:
+    * removes trailing and leading spaces
+    * removes all punctuations
+    * removes double spaces
+    * changes to lowercase
+
+    :param response_text: the response text to be cleaned
+    :return: cleaned response text
+    """
+    response_text = response_text.translate(str.maketrans(string.punctuation, ' ' * len(string.punctuation)))
+
+    while '  ' in response_text:
```

Comment:
Same comment as before - should be a regex.

Reply:
Moved as a common utlity method  for wiki parser and bot in d266d930fbd4885d99aa0bd34300009ef2e3340c